hack/deploy-karmada-agent.sh

```diff
@@ -4,24 +4,66 @@ set -o errexit
 set -o nounset
 
 REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
-KARMADA_APISERVER_KUBECONFIG=${KARMADA_APISERVER_KUBECONFIG:-"/var/run/karmada/karmada-apiserver.config"}
 
-# The host cluster name which used to install karmada control plane components.
-MEMBER_CLUSTER_NAME=${MEMBER_CLUSTER_NAME:-"member3"}
-MEMBER_CLUSTER_KUBECONFIG=${MEMBER_CLUSTER_KUBECONFIG:-"${HOME}/.kube/member3.config"}
-
-source ${REPO_ROOT}/hack/util.sh
 function usage() {
   echo "This script will deploy karmada agent to a cluster."
-  echo "Usage: hack/deploy-karmada-agent.sh"
-  echo "Example: hack/deploy-karmada-agent.sh"
+  echo "Usage: hack/deploy-karmada-agent.sh <KARMADA_APISERVER_KUBECONFIG> <KARMADA_APISERVER_CONTEXT_NAME> <MEMBER_CLUSTER_KUBECONFIG> <MEMBER_CLUSTER_CONTEXT_NAME>"
+  echo "Example: hack/deploy-karmada-agent.sh ~/.kube/karmada.config karmada-apiserver ~/.kube/karmada.config member1"
 }
 
-export REGISTRY="swr.ap-southeast-1.myhuaweicloud.com/karmada"
-export VERSION="latest"
-kind load docker-image "${REGISTRY}/karmada-agent:${VERSION}" --name="${MEMBER_CLUSTER_NAME}"
+if [[ $# -ne 4 ]]; then
+  usage
+  exit 1
+fi
+
+# check kube config file existence
+if [[ ! -f "${1}" ]]; then
+  echo -e "ERROR: failed to get kubernetes config file: '${1}', not existed.\n"
+  usage
+  exit 1
+fi
+KARMADA_APISERVER_KUBECONFIG=$1
+
+# check context existence
+if ! kubectl config use-context "${2}" --kubeconfig="${KARMADA_APISERVER_KUBECONFIG}" > /dev/null 2>&1;
+then
+  echo -e "ERROR: failed to use context: '${2}' not in ${KARMADA_APISERVER_KUBECONFIG}. \n"
+  usage
+  exit 1
+fi
+KARMADA_APISERVER_CONTEXT_NAME=$2
+
+# check kube config file existence
+if [[ ! -f "${3}" ]]; then
+  echo -e "ERROR: failed to get kubernetes config file: '${3}', not existed.\n"
+  usage
+  exit 1
+fi
+MEMBER_CLUSTER_KUBECONFIG=$3
+
+# check context existence
+if ! kubectl config get-contexts "${4}" --kubeconfig="${MEMBER_CLUSTER_KUBECONFIG}" > /dev/null 2>&1;
+then
+  echo -e "ERROR: failed to get context: '${4}' not in ${MEMBER_CLUSTER_KUBECONFIG}. \n"
+  usage
+  exit 1
+fi
+MEMBER_CLUSTER_NAME=$4
+
+source "${REPO_ROOT}"/hack/util.sh
 
-export KUBECONFIG="${MEMBER_CLUSTER_KUBECONFIG}"
+# install agent to member cluster
+CURR_KUBECONFIG=$KUBECONFIG # backup current kubeconfig
+export KUBECONFIG="${MEMBER_CLUSTER_KUBECONFIG}" # switch to member cluster
+kubectl config use-context "${MEMBER_CLUSTER_NAME}"
+
+# load image if member cluster created by kind (if kubeconfig clusters' name contains kind-xxx)
```

Comment:
I'm thinking about why we need to treat `kind` clusters especially?
The reason might be `kind` can't leverage the images on the node where running this script.
Is it the reason?